components/infra-proxy-service/server/roles.go

```diff
@@ -235,7 +223,7 @@ func (s *Server) UpdateRole(ctx context.Context, req *request.UpdateRole) (*resp
 
 // fromAPIToListRoles a response.Roles from a struct of RoleList
 func fromAPIToListRoles(result RoleListResult) []*response.RoleListItem {
-	cl := make([]*response.RoleListItem, result.Total)
+	cl := make([]*response.RoleListItem, len(result.Rows))
```

Comment:
Observe inconsistency in chef server data:
```
knife raw /search/role
{
  "total": 4,
  "start": 0,
  "rows": [

  ]
}
```
To fix this calculating the rows length.